### DISCUSSION TITLE
Does Wasserstein distance/metric follow triangle inequality in ot.emd2 implementation?

### DISCUSSION THREAD

**Discussion by ogencoglu** [PythonOT/POT#436] [Q&A] [answered]
This is a question and not an issue.

I can find sources that show Wasserstein distance indeed follow triangle inequality but does Wasserstein distance/metric implemented by `ot.emd2` follow triangle inequality as is?

**Comment by rflamary** [comment, marked as answer]
Hello emd2 returns the linear OT loss:
- it is equal to $W_1$  when cost matrix M contains euclidean distances (and respect triangular inequality)
- it is equal to $W_2^2$  when cost matrix M contains suqared euclidean distances (and respect triangular inequality whne a square root is added)
